lib/mint/web_socket/upgrade_failure_error.ex

```diff
@@ -3,6 +3,10 @@ defmodule Mint.WebSocket.UpgradeFailureError do
   An error representing a failure to upgrade protocols from HTTP to WebSocket
   """
 
+  @type t() :: %__MODULE__{
+          status_code: pos_integer,
```

Comment:
We can re-use Mint's status-code type here: [`t:Mint.Types.status/0`](https://hexdocs.pm/mint/Mint.Types.html#t:status/0)